### PR TITLE
Add metrics to ZipkinSpanExporter

### PR DESCRIPTION
**Pull request by Donnerbart**
This is a follow-up to https://github.com/open-telemetry/opentelemetry-java/pull/4487 to complete the metrics for span exporters story.

I'm not sure if you like the distinct transport name for HTTP/JSON encoding in the `instrumentationScopeName`. This will change the existing metric names for `OkHttpExporter` when JSON encoding is used. I made that part an extra commit which can easily be dropped, if that change is not wanted.

### CODE REVIEW

**Comment by Donnerbart** [comment]
I think the field `baseLogger` and the method `produceLocalIp()` in `ZipkinSpanExporter` should be `private`. This change technically breaks binary compatibility, but I think those should never have been exposed. Is there a way to do this change without failing the `exporters:zipkin:jApiCmp` check or can't this be changed anymore?

**Comment by jkwatson** [comment]
> I think the public field `baseLogger` and the package-private method `produceLocalIp()` in `ZipkinSpanExporter` should be `private`. This change would break the binary compatibility, but I think it's very unlikely that someone is using these anyway. Is there a way to do this change without failing the `exporters:zipkin:jApiCmp` check or can't this be changed anymore?

Unfortunately, we're stuck with that mistake. :(

**Comment by jack-berg** [comment]
Sorry for not responding sooner. I've been hesitant to accept this because it introduces a new transitive dependency for the `:exporters:zipkin` module on `com.squareup.okhttp3:okhttp`. More generally, the `:exporters:otlp:common` artifact is becoming a home for utility methods used by a variety of exporters, which feels sloppy.

Maybe we can move the truly common classes to a new module `:exporters:common`, and merge remaining shared otlp utilities into `:exporters:otlp:all`. Thoughts @anuraaga / @jkwatson?

**Comment by jack-berg** [comment]
It shades in okhttp [io.zipkin.reporter2:zipkin-sender-okhttp3](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/zipkin/build.gradle.kts#L18).

**Comment by jack-berg** [comment]
This is unblocked now that #4575 is merged. I've pushed a commit to rebase. Thanks @Donnerbart!

**Review by jack-berg — APPROVED** [review]

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java [hunk 1]
```diff
@@ -58,21 +58,39 @@ public void addFailed(long value) {
     exported.add(value, failedAttrs);
   }
 
-  /** Create an instance for recording gRPC exporter metrics. */
+  /**
+   * Create an instance for recording exporter metrics under the meter "{@code
+   * io.opentelemetry.exporters.}" + {@code exporterName} + "-grpc".
```
Comment (jkwatson) [inline exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java:63]: does this javadoc work???

Comment (jack-berg) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java]: It certainly leaves room for improvement! Fixed.

**Comment by jack-berg** [comment]
@jkwatson any additional thoughts before I merge this?

**Review by jkwatson — APPROVED** [review]
🚢

**Pull request merged**